### PR TITLE
Bulk production queue

### PR DESCRIPTION
**Pull request by MunWolf**
This adds a new production queue.

It allows the player to place an order and get them in a bulk.

Its possible to define a minimum and maximum amount of units in each order.
You can also define a static production time or use a dynamic one.

You currently cannot do multiple orders at a time but will have to wait for the current one to finish.

This needs a UI that implements the Finalize Production order since currently it just adds the units to an order as soon as it can.

P.S. The production time does not include the time it takes for the starport transport to arrive.

### CODE REVIEW

**Comment by pchote** [comment]
We'll still want this, but we can't do much more until you (or someone else) rebases and fixes it up.  Closing for now as part of a much-needed pull queue cleanup.

**Comment by Bynnar18** [comment]
I would really like to see this looked at again. Beyond using it for a starport, it could be useful in a variety of applications.

File: OpenRA.Mods.RA/Cargo.cs [hunk 1]
```diff
@@ -190,6 +190,11 @@ public string VoicePhraseForOrder(Actor self, Order order)
 		public Actor Peek(Actor self) { return cargo[0]; }
 
 		public Actor Unload(Actor self)
+		{
+			return this.Unload(self, null);
+		}
+
+		public Actor Unload(Actor self, RallyPoint rally)
```
Comment (Mailaender) [inline OpenRA.Mods.RA/Cargo.cs]: `RallyPoint rally` is never  used.

**Comment by MunWolf** [comment]
Man I forgot I even did this, its been like 3 years now.
Not sure if I will have the time / drive to start programming for OpenRA again but anyone is welcome to pick this up if they want to.

**Comment by Mailaender** [comment]
I tried to rebase it https://github.com/Mailaender/OpenRA/tree/bulk-production but couldn't get it to work.

**Comment by MunWolf** [comment]
Like I said, this is very old code so it is probably completly broken at this point, not sure if any of the other PRs that came after me tackled exactly this problem but you might want to look at those.

**Comment by pchote** [comment]
#15018 tackles a lot of the underlying plumbing. The amount of code needed to implement this should go down significantly once that is merged

**Pull request closed without merging**